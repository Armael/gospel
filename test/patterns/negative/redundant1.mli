val f : int -> int
(*@ y = f x
    ensures match x with
    | _ -> true
    | 1i -> false *)
