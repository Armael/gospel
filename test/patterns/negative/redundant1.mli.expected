
*******************************
********** Parsed file ********
*******************************
val f : int -> int[@@gospel
                    {| y = f x
    ensures match x with
    | _ -> true
    | 1i -> false |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

val f : int -> int
(*@ y = f x
    ensures ...
     *)
File "redundant1.mli", line 3, characters 12-58:
Error: The pattern-matching is redundant.
       Here is a case that is unused:
         1i.
